### PR TITLE
[12.0][FIX] dms: Fix preview widget

### PR DESCRIPTION
**Pull request by etobella**
This fixes the behavior of the previewer

### CODE REVIEW

File: dms/static/src/js/fields/preview.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ odoo.define("dms.preview", function(require) {
                 "/" +
                 this.fieldName +
                 "/" +
-                "datas" +
+                this.activeAttachment.name +
```
Comment (pedrobaeza) [inline dms/static/src/js/fields/preview.js]: Are you sure about this?

File: dms/static/src/xml/views.xml [hunk 2]
```diff
@@ -57,24 +57,44 @@
...
-    <t t-name="FieldBinaryPreview" t-extend="DocumentViewer">
-        <t t-jquery=".o_document_viewer_content_call" t-operation="replace">
-            <t
-                class="o_document_viewer_content_call"
-                t-call="FieldBinaryPreview.Content"
-            />
-        </t>
+    <t t-name="FieldBinaryPreview">
+        <div
+            class="modal o_modal_fullscreen"
```
Comment (pedrobaeza) [inline dms/static/src/xml/views.xml]: Why this change?

File: dms/static/src/js/fields/preview.js (hunk 1 above)
Comment (etobella) [reply dms/static/src/js/fields/preview.js]: Yes, I am using https://github.com/odoo/odoo/blob/12.0/addons/web/controllers/main.py#L1029

Comment (pedrobaeza) [reply dms/static/src/js/fields/preview.js]: But is that route trustable? May there be conflicts with the name (due to slug)?

Comment (pedrobaeza) [reply dms/static/src/js/fields/preview.js]: Isn't better to use the route without that part?

File: dms/static/src/xml/views.xml [hunk 3]
```diff
@@ -57,24 +57,21 @@
...
             </div>
         </t>
     </t>
     <t t-name="FieldBinaryPreview" t-extend="DocumentViewer">
-        <t t-jquery=".o_document_viewer_content_call" t-operation="replace">
-            <t
-                class="o_document_viewer_content_call"
-                t-call="FieldBinaryPreview.Content"
-            />
+        <t t-jquery="[t-call='DocumentViewer.Content']" t-operation="replace">
```
Comment (pedrobaeza) [inline dms/static/src/xml/views.xml:77]: A replace operation is not usually advisable. Isn't there any other option?

Comment (etobella) [reply dms/static/src/xml/views.xml:77]: I know that it's a replace, but the new element inherits from DocumentViewer.Content, It was the only way I found easy enough in order to change the logic :thinking:

Comment (pedrobaeza) [reply dms/static/src/xml/views.xml:77]: OK if you don't see a better approach, but always remember to add a comment explaining the whys for better later comprehension.

File: dms/static/src/js/fields/preview.js (hunk 1 above)
Comment (etobella) [reply dms/static/src/js/fields/preview.js]: With your proposal the downloaded file will be named as datas.pdf or datas.jpg. 
With mine, the name of the file, will be the name of the file on odoo.

Not a big issue for me, I can change it if you think is better :wink:

Comment (pedrobaeza) [reply dms/static/src/js/fields/preview.js]: Ah, OK. Is this handling correctly accented chars and so on?

Comment (etobella) [reply dms/static/src/js/fields/preview.js]: I tried several options and I found one that could give problems, so I am checking again :wink:

Comment (pedrobaeza) [reply dms/static/src/js/fields/preview.js]: OK, let me know :smiley:

Comment (pedrobaeza) [reply dms/static/src/js/fields/preview.js]: Maybe a call to `slug` over that name (is available at this level? or you need a computed field for that?)

Comment (etobella) [reply dms/static/src/js/fields/preview.js]: Well, I used the URLSearchParams in order to remove the problem, so, now it should work

**Comment by pedrobaeza** [comment]
Great, for testing this functionally, what should I look before the patch and after?

**Comment by etobella** [comment]
imagen
 This button was not working properly

**Comment by pedrobaeza** [comment]
I get a weird behavior using arrow keys in the preview:

Peek 28-08-2020 20-19

**Comment by etobella** [comment]
Yes, thanks, I have found the issue, now it should be solved :wink:

**Review by pedrobaeza — APPROVED** [review]
Is this applicable to v13?

**Comment by etobella** [comment]
Yes, it is the last part :disappointed:

**Comment by pedrobaeza** [comment]
Let's merge this for fixing the problem:

/ocabot merge patch
